edward/inferences/hmc.py

```diff
@@ -152,18 +152,23 @@ def _log_joint(self, z_sample):
     return log_joint
 
 
-def leapfrog(z_old, r_old, step_size, log_joint):
+def leapfrog(z_old, r_old, step_size, log_joint, n_steps):
   z_new = {}
   r_new = {}
 
+  for key in z_old:
```

Comment:
i think these lines can be condensed with the the copy method in dict's, e.g., `z_new = z_old.copy()` and `r_new = r_old.copy()`.